### PR TITLE
chore(ci): update github lint and cache workflow

### PR DESCRIPTION
**Pull request by robert-zaremba**
## Description

While working on the cosmos-v046 branch, I found that our golangci-lint is not working correctly. Here is the fix.

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] added appropriate labels to the PR
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/umee-network/umee/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items._

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

**Comment by RafilxTenfen** [comment]
The lint issues are going to be solved on this PR as well?

**Comment by robert-zaremba** [comment]
> The lint warnings are going to be solved on this PR as well?

Only errors. We can fix warnings in a subsequent PR.

**Review by RafilxTenfen — APPROVED** [review]

**Review by toteki — APPROVED** [review]

File: price-feeder/oracle/oracle.go [hunk 1]
```diff
@@ -276,6 +276,7 @@ func GetComputedPrices(
 	providerPairs map[provider.Name][]types.CurrencyPair,
 	deviations map[string]sdk.Dec,
 ) (prices map[string]sdk.Dec, err error) {
+
```
Comment (adamewozniak) [inline price-feeder/oracle/oracle.go:279]: Is this a new lint rule?

File: x/oracle/abci.go [hunk 2]
```diff
@@ -10,7 +10,7 @@ import (
 	sdk "github.com/cosmos/cosmos-sdk/types"
 )
 
-// IsPeriodLastBlock returns true if we are at the last block of the period
+// BADIsPeriodLastBlock returns true if we are at the last block of the period
```
Comment (adamewozniak) [inline x/oracle/abci.go:13]: Where are these "BAD", "XXX" comment pre/suffixes coming from? I don't see them in main

**Review by adamewozniak — APPROVED** [review]
👍 lgtm, although idk why golangci-price-feeder isn't running

File: price-feeder/oracle/oracle.go (hunk 1 above)
Comment (robert-zaremba) [reply price-feeder/oracle/oracle.go:279]: this file was not autoformatted with gofmt. I just opened it and saved it.

File: x/oracle/abci.go (hunk 2 above)
Comment (robert-zaremba) [reply x/oracle/abci.go:13]: No idea :D, let me see in commits.

Comment (robert-zaremba) [reply x/oracle/abci.go:13]: I added it to test the linter.

Comment (robert-zaremba) [reply x/oracle/abci.go:13]: removing

**Comment by robert-zaremba** [comment]
> idk why golangci-price-feeder isn't running

It is running, I tested it. Revive doesn't work for some reason, and created follow up task for that.

**Comment by robert-zaremba** [comment]
The problem with `golangci-price-feeder` is that I removed it and merged it into the lint job. We need to change setting because it's set as a required. Since it's merged into the `Lint` job, we need to remove that entry.

**Pull request merged**